druid-shell/src/platform/mac/window.rs

```diff
@@ -758,6 +762,43 @@ extern "C" fn draw_rect(this: &mut Object, _: Sel, dirtyRect: NSRect) {
     }
 }
 
+fn run_deferred(this: &mut Object, view_state: &mut ViewState, op: DeferredOp) {
+    match op {
+        DeferredOp::SetSize(size) => set_size_deferred(this, view_state, size),
+        DeferredOp::SetPosition(pos) => set_position_deferred(this, view_state, pos),
+    }
+}
+
+fn set_size_deferred(this: &mut Object, _view_state: &mut ViewState, size: Size) {
+    unsafe {
+        let window: id = msg_send![this, window];
+        let current_frame: NSRect = msg_send![window, frame];
+        let mut new_frame = current_frame;
+
+        // TODO: maintain druid origin (as mac origin is bottom left)
+        new_frame.origin.y -= size.height - current_frame.size.height;
+        new_frame.size.width = size.width;
+        new_frame.size.height = size.height;
+        let () = msg_send![window, setFrame: new_frame display: YES];
+    }
+}
+
+fn set_position_deferred(this: &mut Object, _view_state: &mut ViewState, position: Point) {
+    unsafe {
+        // TODO this should be the max y in orig mac coords
```

Comment:
ditto, not sure if this is an issue?

Reply:
It is if there is a weird monitor layout. Secondary monitors above And below the primary or something like that

Reply:
I made the comment more explicit. I'll fix it when I can set up this weird monitor arrangement to test it or someone else can.